### PR TITLE
Install azure cli and powershell on azure linux 3.0

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Review by mthalman — COMMENTED** [review]
`azure-cli` doesn't exist in Azure Linux package feed.

**Comment by mmitche** [comment]
> `azure-cli` doesn't exist in Azure Linux package feed.

That...is not ideal.

**Review by mthalman — APPROVED** [review]

File: src/azurelinux/3.0/net8.0/crossdeps/amd64/Dockerfile [hunk 1]
```diff
@@ -25,3 +25,22 @@ RUN tdnf update -y && \
...
+# Install powershell from the rhel8 MS repo, and azure-cli using PIP
+
+# Install powershell from the Microsoft repository. 
+RUN curl -sSL -O https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm && \
+    rpm -i packages-microsoft-prod.rpm && \
+    rm -f packages-microsoft-prod.rpm && \
+    tdnf update -y && \
+    tdnf install -y --setopt tsflags=nodocs \
+         powershell && \
+    tdnf clean all
```
Comment (mthalman) [inline src/azurelinux/3.0/net8.0/crossdeps/amd64/Dockerfile:39]:
```suggestion
    rm /etc/yum.repos.d/microsoft-prod.repo && \
    tdnf clean all
```

File: src/azurelinux/3.0/net9.0/crossdeps/amd64/Dockerfile [hunk 2]
```diff
@@ -25,3 +25,22 @@ RUN tdnf update -y && \
...
+# Install powershell from the rhel8 MS repo, and azure-cli using PIP
+
+# Install powershell from the Microsoft repository. 
+RUN curl -sSL -O https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm && \
+    rpm -i packages-microsoft-prod.rpm && \
+    rm -f packages-microsoft-prod.rpm && \
+    tdnf update -y && \
+    tdnf install -y --setopt tsflags=nodocs \
+         powershell && \
+    tdnf clean all
```
Comment (mthalman) [inline src/azurelinux/3.0/net9.0/crossdeps/amd64/Dockerfile:39]:
```suggestion
    rm /etc/yum.repos.d/microsoft-prod.repo && \
    tdnf clean all
```

**Comment by mmitche** [comment]
Thanks.

**Pull request merged**